client/components/Editor/schemas/reference.ts

```diff
@@ -0,0 +1,65 @@
+import classNames from 'classnames';
+import { Hooks } from '@pubpub/prosemirror-reactive/src/store/types';
+
+export default {
+	reference: {
+		isLeaf: true,
+		inline: true,
+		reactive: true,
+		selectable: true,
+		group: 'inline',
+		attrs: {
+			id: { default: null },
+			targetId: { default: null },
+		},
+		reactiveAttrs: {
+			label: function(this: Hooks, node) {
+				const { targetId } = node.attrs;
+				const { blockNames } = node.type.spec.defaultOptions;
```

Comment:
How would you feel about storing this in the `documentState` argument of the (pubpub-side) `reactive.ts` plugin wrapper, and then retrieving it with the `useDocumentState` hook? The name is a bit of a misnomer because hooks won't actually re-run based on mutations to (yet), but it's the replacement I had in mind for the use of `node.type.spec.defaultOptions`.

Reply:
I think it's a great idea, I'll work on that now.